modules/dop/services/issue/issue.go

```diff
@@ -692,18 +700,147 @@ func (svc *Issue) UpdateIssue(req apistructs.IssueUpdateRequest) error {
 	return nil
 }
 
-func (svc *Issue) AfterIssueChildrenUpdate(id uint64) error {
-	parents, err := svc.db.GetIssueParents(id, []string{apistructs.IssueRelationInclusion})
-	if err != nil {
+type issueChildrenUpdated struct {
+	id       uint64
+	stateOld apistructs.IssueStateBelong
+	stateNew apistructs.IssueStateBelong
+}
+
+func (svc *Issue) AfterIssueChildrenUpdate(u *issueChildrenUpdated) error {
+	if u == nil {
+		return fmt.Errorf("issue children update config is empty")
+	}
+	parents, err := svc.db.GetIssueParents(u.id, []string{apistructs.IssueRelationInclusion})
+	if err != nil || len(parents) != 1 {
```

Comment:
it is quite confusion, I suggest to change to:
```
if err != nil {
  return err
}
if l := len(parents); l != 1 {
  if l > 1 {
    // log warning
  }
  return nil
}
```

Reply:
done